### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by NduatiKagiri**
In this branch I:

- Created `Classroom` class
- Created the `has-many/belongs-to` relationship between `Classroom` and `Student`
- Created `Book` class
- Created `Rental` class
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (t-yanick) [inline book.rb]: - [ ] The `add_rental` method takes person and date as parameters, the book itself should also be sent as a parameter to create the new rental achieving the 'has-many' association. Something like so: 

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

Please kindly fix that.

File: person.rb [hunk 2]
```diff
@@ -20,6 +22,11 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
+    @rentals.push(rental)
+    rentals.person = self
+  end
```
Comment (t-yanick) [inline person.rb]: - [ ] The `add_rental` method takes book and date as parameters, the person itself should also be sent as a parameter to create the new rental achieving the 'has-many' association. Something like so: 

```ruby
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
```

Please kindly fix that.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @NduatiKagiri, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle:
success

## Hey @NduatiKagiri  :wave:

Great work on implementing the changes requested by the previous reviewer. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:
:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Descriptive PR Summary
:heavy_check_mark: Following the GitFlow
:heavy_check_mark: Implemented Associations

# Optional


_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**